hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -417,9 +417,16 @@ public SingleTransactionRecordBuilderImpl transactionID(@NonNull final Transacti
     @NonNull
     public SingleTransactionRecordBuilderImpl syncBodyIdFromRecordId() {
         final var newTransactionID = transactionID;
-        final var body =
-                inProgressBody().copyBuilder().transactionID(newTransactionID).build();
-        this.transaction = SingleTransactionRecordBuilder.transactionWith(body);
+        var body = inProgressBody().copyBuilder().transactionID(newTransactionID);
+
+        // for preserving correct data cases
+        if (transaction.body() != null && transaction.body().hasNodeStakeUpdate()) {
+            body.nodeStakeUpdate(transaction.body().nodeStakeUpdate());
+        } else if (transaction.body() != null && transaction.body().hasCryptoCreateAccount()) {
+            body.cryptoCreateAccount(transaction.body().cryptoCreateAccount());
+        }
```

Comment:
Why do we need this ?

Reply:
@Neeharika-Sompalli for preserving correct data case, without this `TransactionBodyValidator` will fail with 
"Invalid TransactionBody type HederaFunctionality.NONE with record:..."

Reply:
Which test case helps us demonstrate this failure. Adding this case only for these two doesnt seem correct. Is that because they are synthetic transactions that dont have special Record interfaces that implement `SingleTransactionRecord` ?